test/t/io/test_bzip2.cpp

```diff
@@ -7,13 +7,19 @@
 
 #include <string>
 
+static void read_from_decompressor(int fd) {
+    osmium::io::Bzip2Decompressor decomp(fd);
+    decomp.read();
+    decomp.close();
+}
+
 TEST_CASE("Invalid file descriptor of bzip2-compressed file") {
-    REQUIRE_THROWS_AS(osmium::io::Bzip2Decompressor{-1}, std::system_error);
+    REQUIRE_THROWS_AS(read_from_decompressor(-1), std::exception);
```

Comment:
If you don't check for a specific exception, just using `REQUIRE_THROWS()` is better. Same issue below.